src/resolver/generics.rs

```diff
@@ -178,6 +167,49 @@ impl<'i> TypeAnnotator<'i> {
         }
     }
 
+    fn find_or_create_datatype(
+        &mut self,
+        member_name: &str,
+        generics: &HashMap<String, String>,
+    ) -> String {
+        match self.index.find_effective_type_info(member_name) {
+            Some(DataTypeInformation::Generic { generic_symbol, .. }) => {
+                // this is a generic member, so lets see what it's generic symbol is and translate it
+                generics
+                    .get(generic_symbol)
+                    .map(String::as_str)
+                    .unwrap_or_else(|| member_name)
+                    .to_string()
+            }
+            Some(DataTypeInformation::Pointer {
```

Comment:
for better readability can you provide some explanation and an example in a comment where this match would hit?

does this problem exist for POINTER TO T / ARRAY OF T as well?

Reply:
In theory yes. My idea was only to solve it for cases we currently know we need because it would be more complex when it comes to the user defining a POINTER TO T or ARRAY OF T instead of the ByRef just deriving that.
I would suggest we do this in a follow-up issue.

Reply:
kk fine for me